tar/read.c

```diff
@@ -264,8 +266,7 @@ read_archive(struct bsdtar *bsdtar, char mode, struct archive *writer)
 		}
 		if (r == ARCHIVE_FATAL)
 			break;
-		const char *p = archive_entry_pathname(entry);
-		if (p == NULL || p[0] == '\0') {
+		if (archive_entry_pathname(entry) == NULL || archive_entry_pathname(entry)[0] == '\0') {
```

Comment:
Oh, yes, now I remember.  VS2012 does not support C99, so all variable declarations must be at the beginning of a block.

The "right" fix here is to put
   const char *p;
at the top of the block and change the original declaration to just assignment:
   p = archive_entry_pathname(entry);